content/blog/2019-10-26-lazy-code-motion/index.md

```diff
@@ -0,0 +1,206 @@
+predecessor or successor block. This could have performance benefits because it
+would reduce the number of jumps. Similarly the pretty-printer for CFGs
+does not omit jumps where fall-through would work--this seems like a minor
+detail but could have performance or code size implications.
+
+# Evaluation
+I instrumented the interpreter to count operations and computations.
+A computation is any instruction that computes the value of an expression. An
+operation is any instruction at all, including control flow and return
+instructions. Many of the programs in the `examples` and `test` directories are
+either a single basic block or include no computations, so I excluded them from
+this evaluation because the optimization doesn't do anything to them.
+
+```
++-----------------------------------------------------------------------------------------------------------------+
+|program                      |ops (before lcm)|computations (before lcm)|ops (after lcm)|computations (after lcm)|
++-----------------------------------------------------------------------------------------------------------------+
+|test/lcm/dont-hoist-thru-loop|               8|                        1|             11|                       1|
++-----------------------------------------------------------------------------------------------------------------+
+|test/lcm/two-blocks          |               5|                        1|              8|                       1|
++-----------------------------------------------------------------------------------------------------------------+
+|test/lcm/hoist-thru-loop     |             412|                      303|            617|                     203|
++-----------------------------------------------------------------------------------------------------------------+
+|test/lcm/simple              |            1217|                      906|           1925|                     605|
++-----------------------------------------------------------------------------------------------------------------+
+|examples/lvn_test/nonlocal   |               7|                        3|             12|                       3|
++-----------------------------------------------------------------------------------------------------------------+
+|examples/dom_test/loopcond   |             117|                       46|            165|                      46|
++-----------------------------------------------------------------------------------------------------------------+
+|examples/df_test/cond        |               9|                        1|             12|                       1|
++-----------------------------------------------------------------------------------------------------------------+
+|examples/df_test/fact        |              62|                       25|             89|                      25|
++-----------------------------------------------------------------------------------------------------------------+
+```
```

Comment:
FWIW, I think our Markdown processor supports [tables](https://github.github.com/gfm/#tables-extension-), so you can turn this into a legit table if you want.